demo/client-filter.html

```diff
@@ -0,0 +1,155 @@
+    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.1.2/js/bootstrap.min.js" integrity="sha384-o+RDsa0aLu++PJvFqy8fFScvbHFLtbvScb8AjopnFD+iEQ7wo/CG0xlczd+2O/em" crossorigin="anonymous"></script>
+    <!-- Place this tag in your head or just before your close body tag. -->
+    <script async defer src="https://buttons.github.io/buttons.js"></script>
+    <link rel="stylesheet" href="https://unpkg.com/leaflet@1.3.2/dist/leaflet.css" />
+    <script src="https://unpkg.com/leaflet@1.3.2/dist/leaflet.js"></script>
+    <link rel="stylesheet" href="style.css">
+    <script>
+        "use-strict"
+
+        document.addEventListener('DOMContentLoaded', function pageLoaded() {
+            var map = L.map('map', {
+                center: [39.97, -75.15], zoom: 11
+            });
+
+            // add basemap
+            var basemap = L.tileLayer('https://korona.geog.uni-heidelberg.de/tiles/roadsg/x={x}&y={y}&z={z}', {
+                maxZoom: 19,
+                attribution: 'Imagery from <a href="http://giscience.uni-hd.de/">GIScience Research Group @ University of Heidelberg</a> &mdash; Map data &copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>'
+            });
+            map.addLayer(basemap)
+
+            // dynamically set path based on host
+            var PROD_PATH = 'https://d3hvvzen5hf4hb.cloudfront.net/';
+            var DEV_PATH = 'http://localhost:3000/';
+            var TILESERVER_PATH;
+            var host = window.location.hostname;
+            if (!host || host === 'localhost') TILESERVER_PATH = DEV_PATH;
+            else TILESERVER_PATH = PROD_PATH;
+
+            // add filtered layers
+            function newLayer (filters) {
+                return L.tileLayer(TILESERVER_PATH + 'tile/{z}/{x}/{y}.png?config=data-type-example&layers=' + JSON.stringify(filters))
+            }
+
+            var EXAMPLE_PATH = 'https://yourtiles.cloudfront.net/tile/{z}/{x}/{y}.png?layers='
+
+            var INITIAL_QUERY = [
+                {
+                    name: 'pwd_parcels',
+                    mode: 'AND',
+                    filters: [
+                        {
+                            col: 'owner2',
+                            val: 'OF PHILADELPHIA',
+                        },
+                        {
+                            col: 'gross_area',
+                            op: '>',
+                            val: '3000'
+                        }
+                    ]
+                }
+            ]
+
+            // keep track of active layers so
+            var currentSelection;
+            document.getElementById("btnUpdateQuery").addEventListener('click', function selectorMouseUp() {
+                // remove old layer
+                map.removeLayer(currentSelection);
+
+                // get new layer schema
+                var layerObj;
+                try {
+                    layerObj = JSON.parse(document.getElementById('querybox').value);
```

Comment:
I tried to do a `LIKE` query and it crashed because the `%` wildcards confused escaping.  Escaping them myself worked, i.e.:
```json
[
  {
    "name": "pwd_parcels",
    "mode": "AND",
    "filters": [
      {
        "col": "owner2",
        "val": "%25PHILA%25",
"op": "LIKE"
      },
      {
        "col": "gross_area",
        "op": "<",
        "val": "3000"
      }
    ]
  }
]
```
Not sure if it's worth the trouble to try to escape the `val` values here in the demo, but it might be worth noting that gotcha in the documentation.

Reply:
It shouldn't be too big of a deal to add a catch for that case in. Are there any other situations like this that you can imagine occurring, with the differences in which HTTP/JSON/PSQL escape data?